e2e/test/scenarios/visualizations-tabular/object_detail.cy.spec.js

```diff
@@ -101,30 +101,28 @@ describe("scenarios > question > object details", { tags: "@slow" }, () => {
             alias: "Products",
           },
         ],
-        limit: 5,
+        limit: 2,
       },
     };
 
     createQuestion(questionDetails, { visitQuestion: true });
+    cy.findByTestId("question-row-count").should("have.text", "Showing 2 rows");
 
-    cy.log("check click on 1st row");
-
-    cy.get("[data-testid=cell-data]").contains("37.65").realHover();
-    cy.findByTestId("detail-shortcut").findByRole("button").click();
-
+    cy.log("Check object details for the first row");
+    cy.findAllByTestId("cell-data").filter(":contains(37.65)").realHover();
+    cy.get("[data-show-detail-rowindex='0']").click();
```

Comment:
I find this selector to be much more reliable and explicit than targeting the test-id that's the same for any row.